scripts/registry-validate-source.py

```
@@ -76,22 +89,70 @@ def _validate_integrations(doc: dict) -> None:
+            if not isinstance(artifact_binaries, list):
+                continue
+            for binary_item in artifact_binaries:
+                if not isinstance(binary_item, dict):
+                    continue
+                binary_name = binary_item.get("name")
+                if isinstance(binary_name, str):
+                    binaries.add(binary_name)
+    seen: set[str] = set()
+    for idx, item in enumerate(shell_init, start=1):
+        if not isinstance(item, dict):
+            raise ValidationError(f"manifest.shell_init[{idx}] must be a table")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)